compiler/resolution/cullOverReferences.cpp

```diff
@@ -1753,11 +1761,75 @@ bool firstPassLowerContextCall(ContextCallExpr* cc)
     // should use 'getter'
     // MPF - note 2016-01: this code does not seem to be triggered
     // in the present compiler.
-    lowerContextCall(cc, USE_CONST_REF);
+    lowerContextCallPreferConstRefValue(cc);
     return true;
   }
 }
 
+static
+void lowerContextCallPreferRefConstRef(ContextCallExpr* cc)
+{
+  CallExpr* refCall = NULL;
+  CallExpr* valueCall = NULL;
+  CallExpr* constRefCall = NULL;
+  choose_type_t which = USE_REF;
+
+  cc->getCalls(refCall, valueCall, constRefCall);
+
+  if (refCall) which = USE_REF;
+  else if(constRefCall) which = USE_CONST_REF;
+
+  lowerContextCall(cc, which);
+}
+
+static
+void lowerContextCallPreferConstRefValue(ContextCallExpr* cc)
+{
+  CallExpr* refCall = NULL;
+  CallExpr* valueCall = NULL;
+  CallExpr* constRefCall = NULL;
+  choose_type_t which = USE_CONST_REF;
+
+  cc->getCalls(refCall, valueCall, constRefCall);
+
+  if(constRefCall) which = USE_CONST_REF;
+  else if(valueCall) which = USE_VALUE;
```

Comment:
Analogously to the above:
`which` is already initialized to `USE_CONST_REF`. Why do it again here?